judge/models/profile.py

```diff
@@ -174,6 +180,14 @@ def generate_api_token(self):
 
     generate_api_token.alters_data = True
 
+    def generate_scratch_codes(self):
+        codes = [pyotp.random_base32(length=16) for i in range(5)]
+        self.scratch_codes = json.dumps(codes)
+        self.save(update_fields=['scratch_codes'])
+        return '\n'.join(codes)
```

Comment:
I'd prefer returning the list of codes, and then looping over them in the template rather than joining them here. (Something about separating UI from backend logic; the frontend could conceivably someday want to show these as bullet points.)